typhos/tests/conftest.py

```diff
@@ -68,6 +72,141 @@ def noapp(monkeypatch):
+        logger.debug(f"\n\nPre test - {num_start} widgets exist:")
+        _dump_widgets(starting_widgets)
+
+    yield
+
+    qtbot_widgets = [ref for ref, _ in getattr(item, "qt_widgets", [])]
+    ending_widgets = get_top_level_widgets()
+
+    app = QtWidgets.QApplication.instance()
+
+    t0 = time.monotonic()
+    while time.monotonic() - t0 < 1.0 and len(_dereference_list(ending_widgets)) > 0:
+        gc.collect()
+        app.processEvents()
+
+    widgets_to_check = list(
+        weakref.ref(w) for w in
+        set(_dereference_list(ending_widgets))
+        - set(_dereference_list(starting_widgets))
+        - set(_dereference_list(qtbot_widgets))
+    )
+    _dump_widgets(widgets_to_check)
+
+    final_widgets = _dereference_list(widgets_to_check)
+
+    cleanup_descriptions = []
+    for widget in list(final_widgets):
+        try:
+            classname = type(widget).__name__
+            name = widget.objectName()
+            if name in {"_typhos_test_suite_ignore_"}:
+                final_widgets.remove(widget)
+                continue
+
+            title = f"title={widget.windowTitle()} name={name}"
+            referrers = gc.get_referrers(widget)
+            desc = str(widget)
+            if isinstance(widget, QtWidgets.QMenu):
+                desc = "Menu with actions: " + ", ".join(repr(action.text()) for action in widget.actions())
+        except RuntimeError:
+            # OK, one last chance for gc
+            referrers = []
+            final_widgets.remove(widget)
+        else:
+            ref_desc = []
+            num_referrers = len(referrers)
+            for ref in referrers:
+                if ref is widget or ref is final_widgets:
+                    num_referrers -= 1
+                    continue
+
+                try:
+                    if isinstance(ref, list):
+                        ref_desc.append(f"list[{len(ref)}]: " + str(ref)[:512])
+                        for sub_ref in gc.get_referrers(ref):
+                            ref_desc.append(f"          --> {sub_ref}")
+                    else:
+                        ref_desc.append(str(ref)[:512])
+                except Exception as ex:
+                    # Everything's destructible! Yeah!
+                    ref_desc.append(f"(exception) {ex}")
+            desc = f"{classname} {desc} {title} referrers={num_referrers}: "
+            cleanup_descriptions.append(
+                "\n".join((desc, "\n    -> ".join([""] + ref_desc)))
+            )
+        referrers.clear()
+
+    cleanup_text = (
+        f"Not all widgets were cleaned up during {item.name}:\n"
+        + "\n".join(sorted(cleanup_descriptions))
+    )
+    failure_text = f"{item.nodeid}: {cleanup_text}"
+
+    if final_widgets:
+        if all(isinstance(widget, QtWidgets.QMenu) for widget in final_widgets):
+            logger.error("%s: Top level QMenu widgets were not cleaned up. Not failing the test suite.", item.nodeid)
```

Comment:
This actually only happens in *one test*!
`test_take_top_level_widget_screenshots`
It doesn't make sense to me why this is the case.
Still, this escape hatch is what is allowing the test suite to pass for all versions of Python.